From b13f5286ce21d86985339da2b20901a9a6596144 Mon Sep 17 00:00:00 2001
From: NBruderman <nbruderman@gmail.com>
Date: Thu, 11 Jun 2015 15:08:49 +0300
Subject: [PATCH 10/13] Camera2: Fix Undo button behaviour

The undo behaviour depends on if the camera present secure settings
or not. If secure, it should have 1 placeholder. 0 in other cases.

Repro (for non secure cameras):
1. Leave only 3 pictures on the album.
2. Delete one picture, it will work fine.
3. Leave only 2 pictures on the album.
4. Try to delete, and you will go straight to the camera,
without the undo pannel showing up.

Change-Id: I360076381867a74628ffa492065ccf2ca53c72d6
---
 src/com/android/camera/CameraActivity.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index 35a085359..82640c285 100644
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -1350,7 +1350,8 @@ public class CameraActivity extends QuickActivity
 
     private void removeItemAt(int index) {
         mDataAdapter.removeAt(index);
-        if (mDataAdapter.getTotalNumber() > 1) {
+        final int placeholders = mSecureCamera ? 1 : 0;
+        if (mDataAdapter.getTotalNumber() > placeholders) {
             showUndoDeletionBar();
         } else {
             // If camera preview is the only view left in filmstrip,
-- 
2.17.1

